extensions/cornerstone/src/utils/promptHydrationDialog.ts

```diff
@@ -0,0 +1,189 @@
+export type HydrationCallback = (params: any) => Promise<boolean>;
+
+export const HydrationType = {
+  SEG: 'SEG',
+  SR: 'SR',
+  RTSTRUCT: 'RTSTRUCT',
+} as const;
+
+export interface HydrationDialogProps {
+  servicesManager: AppTypes.ServicesManager;
+  viewportId: string;
+  displaySet: AppTypes.DisplaySet;
+  preHydrateCallbacks?: HydrationCallback[];
+  hydrateCallback: HydrationCallback;
+  type: string;
+}
+
+export interface HydrationSRResult {
+  userResponse: number;
+  displaySetInstanceUID: string;
+  srSeriesInstanceUID: string;
+  viewportId: string;
+  StudyInstanceUID?: string;
+  SeriesInstanceUIDs?: string[];
+}
+
+const RESPONSE = {
+  NO_NEVER: -1,
+  CANCEL: 0,
+  CREATE_REPORT: 1,
+  ADD_SERIES: 2,
+  SET_STUDY_AND_SERIES: 3,
+  NO_NOT_FOR_SERIES: 4,
+  HYDRATE: 5,
+};
+
+function getCustomizationMessageKey(type: string): string {
+  switch (type) {
```

Comment:
there might be a key issue here as some dialogs are empty, also SR dialog is not what it used to be